scopesim/source/source.py

```diff
@@ -145,89 +145,83 @@ def __init__(self, filename=None, cube=None, ext=0,
                  lam=None, spectra=None, x=None, y=None, ref=None, weight=None,
                  table=None, image_hdu=None, flux=None, **kwargs):
 
-        self.meta = {}
-        self.meta.update(kwargs)
-        # ._meta_dicts contains a meta for each of the .fields. It is primarily
-        # used to set proper FITS header keywords for each field so the source
-        # can be reconstructed from the FITS headers.
-        self._meta_dicts = [self.meta]
+        self._meta = {}
+        self.fields: list[SourceField] = []
 
-        self.fields = []
-        self.spectra = []
+        self._bandpass: SpectralElement | None = None
```

Comment:
It was, until 3.10, where now `| None` is the preferred way. See https://stackoverflow.com/a/69440627 and also https://discuss.python.org/t/clarification-for-pep-604-is-foo-int-none-to-replace-all-use-of-foo-optional-int/26945.

It seems from those discussions, that there's a bit of an opinion-based thing going on 😛 

I personally tend to go with the new `| None`, mainly because it saves me from importing `Optional`, now that we also don't need to import `Union` any more (which is truly obsolete in 3.10, without much arguing I think) and also it's (fractionally) shorter. Also (and I think that was mentioned in one of those links), it is somewhat more explicit in avoiding that the `Optional` is applied to arguments which are "optional" in the sense that they have a default value, but one that is of the same type as the argument (and not `None`). E.g. `def foo(x: int = 5):` should not be typed `def foo(x: Optional[int] = 5):`, that would only be the case for `def foo(x: Optional[int] = None):`, which is equivalent to  `def foo(x: int | None = None):`. So with all of that in mind, I prefer `| None`, because it also conveys more explicitly that the argument (or attribute in this case) can be `None`, as opposed to it being an optional parameter, additional to the above reasons. But I see why someone might prefer `Optional` as well 🙃

Reply:
OK, if `... | None` is the preferred format, the let's go for it. Thanks for the link.

I never interpreted `Optional` as meaning that the argument is optional, but I can see that happening. Thinking about it, I'm now agreeing with the sentiment that `Optional[...]` should not have been allowed in the first place. 

I interpret `x: Optional[int]` as `x` 'is' an integer, or it conceptually doesn't exist at all, while I interpret `x: int | None` as `x` can either 'be' of type integer, or it can 'be' of type NoneType. I usually don't like the latter interpretation because whatever I pass to the function should not 'be' of type `NoneType`, so the first interpretation is what I try to convey with the code. But that is just nonsense, you can only do the latter in Python; `x` cannot not exist.

I would like `int | None` in cases like `def fix_data(data, interpolation_order: int | None = 3)` over `fix_data(data, interpolation_order: Optional[int] = 3)`, e.g. if None means do not interpolate at all (while the default is 3th order), because then it is actually acceptable to pass `interpolation_order=None`.

But well, my interpretation of `Optional[]` is meaningless, because it is something that cannot actually be expressed in the language. So `|` is it then.

Now I want to redo ScopeSim in Rust :-).